UndertaleModLib/Decompiler/Decompiler.cs

```diff
@@ -779,14 +852,21 @@ internal static void DecompileFromBlock(Block block, List<TempVarReference> temp
                         break;
 
                     case UndertaleInstruction.Opcode.Popz:
-                        Expression popped = stack.Pop();
-                        if (!(popped is ExpressionTempVar))
-                            statements.Add(popped);
+                        if (stack.Count > 0)
+                        {
+                            Expression popped = stack.Pop();
+                            if (!(popped is ExpressionTempVar))
+                                statements.Add(popped);
+                        } else
+                        {
+                            statements.Add(new CommentStatement("WARNING: Popz'd an empty stack."));
```

Comment:
I wasn't able to fix the bug, and there was no functional consequence in the specific scenario. Makes sense though.

Here's what the code looks like, with the comments:
```
switch dialog_step {
  case 0:
    {
     if (global.enemy_hp_check <= (global.enemy_hp_max * 0.75)) {
      dialog_step = 1
      dialog_active = 1
      cur_message = 14
      text_location = 3
      text_flipped = 1
      alarm[3] = room_speed
      exit
     }
  // Something was wrong with the stack, reentered the block with 1 variables instead of 0, ignoring
  }
  break
  case 1:
    {
     if (global.enemy_hp_check <= (global.enemy_hp_max * 0.5)) {
      dialog_step = 2
      dialog_active = 1
      cur_message = 16
      text_location = 3
      text_flipped = 1
      alarm[3] = room_speed
      exit
     }
  // Something was wrong with the stack, reentered the block with 1 variables instead of 0, ignoring
  }
 // WARNING: Popz'd an empty stack.
 alarm[1] = room_speed
 // Something was wrong with the stack, reentered the block with 1 variables instead of 0, ignoring
 // Something was wrong with the stack, reentered the block with 1 variables instead of 0, ignoring
}
```

Source:
obj_barghest2_controller_Step_0 - Heartbound Demo

I reviewed the byte-code, and I wasn't sure what was going wrong in the decompiler, but I determined that these issues were related and weren't supposed to leave any actual decompiled code behind.

Reply:
If we can fix the underlying issue, that would be the best solution.

Reply:
Ok, leave it then and open an issue so that we don't forget about this

Reply:
Issue created.